src/workers/event_helpers.rs

```diff
@@ -95,6 +95,37 @@ impl LinkSource for wasmcloud_control_interface::Client {
     }
 }
 
+/// A struct for publishing status updates
+#[derive(Clone)]
+pub struct StatusPublisher<Pub> {
+    publisher: Pub,
+    // Topic prefix, e.g. wadm.status.default
+    topic: String,
+}
+
+impl<Pub> StatusPublisher<Pub> {
+    /// Creates an new status publisher configured with the given publisher that will send to the
+    /// specified topic
+    pub fn new(publisher: Pub, topic: &str) -> StatusPublisher<Pub> {
```

Comment:
Nit: Might want to name the parameter (and the data member of the struct) to be `topic_prefix` so it is just a bit more clear what it is used for